master/docs/developer/stats-service.rst

```diff
@@ -0,0 +1,255 @@
+   * ``port``: The port on which the service is listening.
+   * ``user``: Username of a InfluxDB user.
+   * ``password``: Password for ``user``.
+   * ``db``: The name of database to be used.
+   * ``captures``: A list of instances of subclasses of :py:class:`Capture`. This tells which stats are to be stored in this storage backend.
+   * ``name=None``: (Optional) The name of this storage backend.
+
+   .. py:method:: postStatsValue(self, post_data, series_name, context={})
+
+      :param post_data: A dict of key-value pairs that is sent for storage. The keys of this dict can be thought of as columns in a database and the value is the data stored for that column.
+      :type post_data: dict
+      :param series_name: The name of the time-series for this statistic.
+      :type series_name: str
+      :param context: (Optional) Any other contextual information about the data. Dict of key-value pairs.
+      :type context: dict
+
+      This method constructs a dictionary of data to be sent to InfluxDB in the proper format and sends the data to the influxDB instance.
+
+
+Capture Classes
+---------------
+
+Capture classes are used for declaring the data that needs to be sent to storage backends for storage.
+
+.. py:class:: Capture
+
+   A base class for all capture classes. Not to be used directly. Initlized with the following parameters:
+
+   * ``routingKey``: (tuple) The routing key to be used by :class:`StatsService` to register consumers to the MQ lasyer for the subclass of this class.
+   * ``callback``: The callback registered with the MQ layer for the consumer of a subclass of this class. Each subclass must provide a default callback for this purpose.
+
+   .. py:method:: defaultContext(self, msg):
+
+      A method for providing default context to the storage backends.
+
+   .. py:method:: consumer(self, routingKey, msg):
+
+      Raises ``NotImplementedError``. Each subclass of this method should implement its own consumer.
+      The consumer, when calles (from the mq layer), receives the following arguments:
+
+      * ``routingKey``: The routing key which was registered to the MQ layer. Same as the ``routingKey`` provided to instantiate this class.
+      * ``msg``: The message that was sent by the producer.
+
+.. py:class:: CaptureProperty
+
+   The capture class to use for capturing build properties. It is available in the configuration as ``stats.CaptureProperty``
+
+   It takes the following arguments:
+
+   * ``builder_name``: The name of builder in which the property is recorded.
+   * ``property_name``: The name of property needed to be recorded as a statistic.
+   * ``callback=None``: The callback function that is used by ``CaptureProperty.consumer`` to post-process data before formatting it and sending it to the appropriate storage backends. A default callback is provided for this.
+
+   **The default callback:**
+
+     .. py:function:: default_callback
+
+     Defined in ``CaptureProperty.__init__``. Receives:
```

Comment:
I do not think "Defined in ..." is really necessary.  Those who want to look at the implementation will see where it is defined.  However the list of anticipated parameters is due.